src/metrics.py

```diff
@@ -0,0 +1,55 @@
+import gin
+
+from torchmetrics import Metric
+from sklearn.metrics import (
+    silhouette_score, davies_bouldin_score, calinski_harabasz_score,
+    accuracy_score, f1_score, precision_score, recall_score
+)
+
+class BaseScoreMetric(Metric):
+    def __init__(self, score_func, **kwargs):
+        super().__init__()
+        self.score = 0
+        self.score_func = score_func
+        self.score_func_kwargs = kwargs
+
+    def update(self, *args, **kwargs):
+        self.score = self.score_func(*args, **kwargs, **self.score_func_kwargs)
```

Comment:
As I understood, you are saving only the latest metric. It is okay for training epoch, but doesn't work well with evaluation (validation) step. I propose to either append every score to the list of scores and in the `compute` return the mean and clear the list, or you could have a total sum of metrics and number of updates, so in the end you can just divide total sum by number of updates. Of course, you should clear both sums in the compute before returning result